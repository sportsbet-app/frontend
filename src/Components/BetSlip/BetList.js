import React from 'react'
import Bet from './Bet'

//export betList function with props passed down from Bet Slip 
export default function BetList(props) {
    return (
        <div>
            {props.bet && props.bet.map((bet, index) => <Bet
                bet={bet}
                key={index}
                id={index}
                delete={props.delete}
            />)}
        </div>
    )
}
