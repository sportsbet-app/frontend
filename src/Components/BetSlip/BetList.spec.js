import React from 'react'
//import all dependencies needed
import { render } from '@testing-library/react'
//import renderer for snapshot
import renderer from 'react-test-renderer'
//import the component being tested
import BetList from './BetList'
import BET from './Bet'

//test data to use for the render
const data = [{
  playerCount: 1,
  playerSelectedOption: { value: 'zero' },
  selectedOption: { value: '0' },
  prop:  'testing',
  count: 3
},
{
  playerCount: 2,
  playerOneSelection: { value: 'one' },
  playerTwoSelection: { value: '1' },
  selectedOption: { value: 'won' },
  prop:  'testing',
  count: 3
},
{
  playerCount: 3,
  playerOneSelection: { value: 'two' },
  playerTwoSelection: { value: '2' },
  playerThreeSelection: { value: 'too' },
  selectedOption: { value: 'tu' },
  prop:  'testing',
  count: 3
}]

//test to check that the app will display when called.
describe('<BetList />', () => {
  it('should display', () => {
    //check the render with the test array data 
    render(<BetList bet = {data}/>)
    
  })
  //create a snapshot
  it('matches snapshot', () => {
    const tree = renderer.create(<BetList bet = {data}/>); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });
})