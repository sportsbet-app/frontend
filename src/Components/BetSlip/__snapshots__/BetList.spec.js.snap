// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BetList /> matches snapshot 1`] = `
<div>
  <div>
    //price value of the selected player
    <h1>
       Player : 
      zero
    </h1>
    //run down of the information for the bet. (type of bet, the bet number and the players value )
    <p>
       
      testing
       : 
      3
        
      0
    </p>
    //button to cancel the selected bet
    <button
      onClick={[Function]}
      style={
        Object {
          "color": "black",
        }
      }
    >
       X 
    </button>
  </div>
  <div>
    <h1>
       Player 1 : 
      one
    </h1>
    <h1>
       Player 2 : 
      1
    </h1>
    <p>
      won
    </p>
    <button
      onClick={[Function]}
      style={
        Object {
          "color": "black",
        }
      }
    >
       X 
    </button>
  </div>
  <div>
    <h1>
       Player 1 : 
      two
    </h1>
    <h1>
       Player 2 : 
      2
    </h1>
    <h1>
       Player 3 : 
      too
    </h1>
    <p>
      tu
    </p>
    <button
      onClick={[Function]}
      style={
        Object {
          "color": "black",
        }
      }
    >
       X 
    </button>
  </div>
</div>
`;
